dts/arm/nucleo_f401re.dts

```diff
@@ -27,3 +27,31 @@
 	current-speed = <115200>;
 	status = "ok";
 };
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
```

Comment:
This assumes that the processor is not booting from SRAM. Perhaps this should start at 0x08000000, and the other partitions be changed to match, so it will always work.